### PR TITLE
feat: improve in-cluster-config example

### PR DESCRIPTION
**Pull request by tomplus**
I've updated an example - in_cluster_config.py.

Changes:
- add notice about RBAC which is usually required (for instance https://github.com/kubernetes-client/python/issues/605)
- add infinite loop to the example, IMO it's more convenient when you are going to run it in K8s environment

Thanks.

### CODE REVIEW

File: examples/in_cluster_config.py [hunk 1]
```diff
@@ -14,18 +14,65 @@
...
-    config.load_incluster_config()
-
-    v1 = client.CoreV1Api()
-    print("Listing pods with their IPs:")
-    ret = v1.list_pod_for_all_namespaces(watch=False)
-    for i in ret.items:
-        print("%s\t%s\t%s" %
-              (i.status.pod_ip, i.metadata.namespace, i.metadata.name))
+
+    while True:
```
[the remark this answers is not in the document]
Comment (tomplus) [reply examples/in_cluster_config.py]: I run tests in different way. I prepared a docker image with the installed library and this example as a command. If the pod works forever I can check logs easily, tune RBAC etc.

Your scenario is simpler, I'll remove the loop and I'll add comments about suggested scenario.

Thanks for reviewing it.

**Comment by yliaog** [comment]
thanks for the pr. could you please sqash the commits? it looks good otherwise.

**Pull request merged**